python/ray/llm/_internal/batch/stages/base.py

```diff
@@ -0,0 +1,211 @@
+"""TBA"""
+import logging
+from enum import Enum, auto
+from typing import Any, Dict, AsyncIterator, List, Callable, Tuple
+
+import pyarrow
+from pydantic import BaseModel
+from ray.data.block import UserDefinedFunction
+
+logger = logging.getLogger(__name__)
+
+
+def wrap_preprocess(
+    fn: UserDefinedFunction,
+    input_column: str,
+    carry_over: bool,
+) -> Callable:
+    """Wrap the preprocess function, so that the output schema of the
+    preprocess is normalized to {input_column: ...} (with other input columns
+    if carry_over is True).
+
+    Args:
+        fn: The function to be applied.
+        input_column: The input column name.
+        carry_over: Whether to carry over the output column to the next stage.
+
+    Returns:
+        The wrapped function.
+    """
+
+    def _preprocess(row: dict[str, Any]) -> dict[str, Any]:
+        outputs = {input_column: fn(row)}
+
+        if carry_over:
+            if input_column in row:
+                row.pop(input_column)
+
+            # In the case of column name conflict,
+            # we overwrite the input column with the output column.
+            row.update(outputs)
+        return row
```

Comment:
ok I am not sure if I understand this right. If carry_over is false, you just pass row to the output and won't apply `fn` to it. I don't think that's correct.  you want `row = outputs` if carry_over is false, if it's true you want `row.update(outputs)` in the output. no?